src/common/parse.rs

```diff
@@ -0,0 +1,103 @@
+use core::iter::FromIterator;
+
+use heapless::Vec;
+
+pub trait PackBytes
+where
+    Self: Sized,
+{
+    fn unpack_from_iter(src: impl IntoIterator<Item = u8>) -> Option<Self>;
+    fn unpack_from_slice(src: &[u8]) -> Option<Self> {
+        Self::unpack_from_iter(src.iter().cloned())
+    }
+}
+
+impl PackBytes for u8 {
+    fn unpack_from_iter(src: impl IntoIterator<Item = u8>) -> Option<Self> {
+        src.into_iter().next()
+    }
+}
+
+impl PackBytes for i8 {
+    fn unpack_from_iter(src: impl IntoIterator<Item = u8>) -> Option<Self> {
+        src.into_iter().next().map(|b| b as i8)
+    }
+}
+
+macro_rules! impl_primitive {
+    ($ty:ty, $sz:literal) => {
+        impl PackBytes for $ty {
+            fn unpack_from_iter(src: impl IntoIterator<Item = u8>) -> Option<Self> {
+                let buf: Vec<u8, $sz> = src.into_iter().take($sz).collect();
+                Some(<$ty>::from_le_bytes(buf.into_array().unwrap()))
+            }
```

Comment:
[`array::try_from(bytes)`](https://doc.rust-lang.org/std/primitive.array.html#impl-TryFrom<%26%5BT%5D>-for-%5BT;+N%5D) is better. So, it should be vice versa:

```suggestion
            fn unpack_from_iter(src: impl IntoIterator<Item = u8>) -> Option<Self> {
                let slice = src.into_iter().take(std::mem::size_of::<$ty>());
                Self::unpack_from_slice(slice)
            }
            
            fn unpack_from_slice(src: &[u8]) -> Option<Self> {
                Some(<$ty>::from_le_bytes(src).expect("Source array mismatch with primitive"))
            }
```

Reply:
Oh, I sent a wrong code previously. Updated it.